### ISSUE TITLE
Create flatpak and upload to flathub.org

### ISSUE THREAD

**Issue by Peque** [dino/dino#453]
It would be great to have a flatpak package for easy and secure installation. :blush:

**Comment by bochecha** [comment]
I had opened #343 for this.

Note that even once the Flatpak manifest is merged with #343, that won't be directly suitable for Flathub until Dino makes a release. (Flathub doesn't accept unreleased builds from master)

In any case, see the pull request for the current status.

**Comment by Peque** [comment]
Thanks, subscribed. :blush:

I will leave this opened then. As it seems it will still be an issue even if #343 is merged.

**Comment by ezickler** [comment]
It seams that there is now some kind of beta repo on flathub. Would it be possible to make a beta release an publish it there?
Here is the blog about the new beta repo:
https://blogs.gnome.org/alexl/2019/02/19/changes-in-flathub-land/

**Comment by bochecha** [comment]
@ezickler everything is possible.

However, see #343 for what the Flatpak support is currently blocked on.

**Comment by rugk** [comment]
BTW as Fedora [has dino in it's official packages](https://apps.fedoraproject.org/packages/dino) (link [currently broken](https://github.com/fedora-infra/fedora-packages/issues/432#issuecomment-602037880) though), [I've asked whether they can provide a flatpak package on registry.fedora.org](https://bugzilla.redhat.com/show_bug.cgi?id=1815758), (too).

**Comment by Siosm** [comment]
I have updated the [Flatpak for Dino in a new PR](https://github.com/dino/dino/pull/905) and I have created [a specific branch for submission to Flathub](https://github.com/Siosm/flathub/tree/im.dino.Dino) and a repo for [Fedora](https://pagure.io/travier_flatpaks/dino). Let me know if you would rather have me submitting this directly to Flathub while making sure that you get full access to the final Flathub repo or if you would prefer to do that yourself.

In both cases, the Flatpak should work but has only been lightly tested by me so it might need some changes especially regarding pulseaudio/sound access, filesystem access, etc. I removed:
```
        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
as I'm not sure they are necessary anymore but could be wrong here.

If you use GitHub Actions to build PRs, you may also be interested in [this Flatpak action](https://github.com/marketplace/actions/flatpak-builder) that could build Flatpaks for each PRs making them really easy to test.

**Comment by fiaxh** [comment]
@Siosm The reason why the previous flatpak PR (#343) has not been accepted is that some Dino features won't work in a flatpak. The flatpak issue that Dino depends on is [flatpak/#2301: gpg-agent socket](https://github.com/flatpak/flatpak/issues/2301).

We want to avoid the confusing situation where people obtain the same Dino version from different sources and end ob with a relevantly different feature set.

If you want to have a proper Dino flatpak it would be more helpful to work on [flatpak/#2301: gpg-agent socket](https://github.com/flatpak/flatpak/issues/2301) :+1:

**Comment by travier** [comment]
Oh, thanks a lot for pointing that out and sorry for the noise!

Unfortunately I won't be working on that as I don't use GPG but I will leave everything available if someone else wants to go ahead.

**Comment by tdemin** [comment]
Wrote [another working Flatpak manifest](https://github.com/tdemin/im.dino.Dino) while being perfectly sure no one bothered with making one (as I expected Dino to would already have been on Flathub if someone did) and stumbled upon this issue.

I guess I should run a local repo then.

**Comment by alexanderadam** [comment]
> I guess I should run a local repo then.

Couldn't you just add a PR to [the official repo](https://github.com/flathub/flathub/pulls)?

**Comment by mbridon** [comment]
Hi, I'm @bochecha from this ticket, I had to change my github account (don't ask).

I completely lost interest in a Flatpak build of Dino, because due to unexpected circumstances I ended up hospitalised for almost 2 years and ended up not using XMPP any more.

Sorry about the mess I left behind me, and all the best for your future endeavours ❤️

**Comment by Saroufim** [comment]
Is [this](https://github.com/flatpak/flatpak/issues/2301#issuecomment-551021176) not feasible as a workaround for now?

**Comment by Aman9das** [comment]
How viable is publishing to flathub-beta, so that to access dino one only needs to enable the beta repository.

**Comment by Aman9das** [comment]
Added to flathub beta here: https://github.com/flathub/im.dino.Dino

pgp is disabled obviously

**Comment by selurvedu** [comment]
FYI https://github.com/flatpak/flatpak/issues/2301 was closed as completed via https://github.com/flatpak/flatpak/pull/4958 .

**Comment by tuxayo** [comment]
IIUC, anyone willing to tackle this can update the code from the orphan PR: [343](https://github.com/dino/dino/pull/343) (original author not working on it anymore) and submit another PR.

**Comment by sonnyp** [comment]
Already did https://github.com/dino/dino/pull/1363

**Comment by mar-v-in** [comment]
https://flathub.org/apps/details/im.dino.Dino
